signature-specification.md

```diff
@@ -0,0 +1,519 @@
+
+```
+Base64Url(ProtectedHeader).Base64Url(Payload).Base64Url(
+    Sign(
+        ASCII(Base64Url(ProtectedHeader).Base64Url(Payload))
+    )
+)
+```
+
+</details>
+
+
+### JWS JSON Serialization - [RFC7515 Section 3.2](https://datatracker.ietf.org/doc/html/rfc7515#section-3.2)
+
+Pros:
+
+- JWS is an IETF proposed (RFC 7515) standard with guidance around representation, serialization and verification.
+- Signing libraries available in Go: [square/go-jose](https://github.com/square/go-jose), [lestrrat-go/jwx](https://github.com/lestrrat-go/jwx).
+- Native support for certificate(s) and key id.
+- Native support for signed and unsigned attributes.
+- Supports multiple signatures in the same envelope.
+
+Cons:
+
+- Has a history of vulnerable implementations such as [algorithm substitution attack](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries), bypassing signature verification by setting `alg = none`. Some of the main concerns raised for JWS, can they be addressed with additional layer of validation on top of core libraries.
+- As specification is complex and lacks specificity, there is probability of user shooting themselves in the foot with incorrect implementation, or insufficient validations.
+
+<details><summary>Example (Click to show details)</summary>
+
+`Payload` a.k.a. claims:
+
+```json
+{
+    "notary.v2": {
+        "subjectManifest": {
+            "mediaType": "application/vnd.oci.image.manifest.v1+json",
+            "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+            "size": 16724,
+            "annotations": {
+                "org.acme-rockets.importDate": "2021-04-23T18:25:43.511Z"
+            }
+        },
+        "signedAttrs": {
+            "reserved": {
+                "arl": "http://registry.wabbit.net/arl",
+                "identity": "acme-rockets.io/net-monitor:v1"
+            },
+            "custom": {
+                "buildId": "0001",
+                "imageScanned": "true"
+            }
+        }
+    },
+    "iat": 1624377760337,
+    "exp": 1632153760337
+}
+```
+
+`Payload` contains the subject manifest and other attributes that have to be integrity protected.
+
+- `notary.v2`: Top level node and private claim, encapsulating the notary v2 data. This claim MUST be present.
+- `subjectManifest`: The image manifest that needs to be integrity protected.
+- `signedAttrs`: Contains additional attributes that needs to be integrity protected.
+    - `reserved`: Collection of attributes reserved for notary v2 use such as artifact revocation list (ARL), identity, etc.
+    - `custom`: Collection of user defined attributes such as buildId, imageScanned, etc. Use of this field is OPTIONAL.
+- `iat`: Issued at identifies the time at which the JWT was issued. This claim MUST be present.
+- `exp`: Expiration time identifies the expiration time on or after which the JWT must not be accepted for processing. This claim MUST be present.

```

Comment:
@priteshbandi The field `exp` is inconsistent with the specificaiton.

In the specification, it is OPTIONAL but here is a MUST.
It also conflicts with the section `Artifact Expiry Validation` in https://github.com/notaryproject/notaryproject/discussions/98